tests/integration_test.go

```diff
@@ -127,9 +128,16 @@ func TestXMRTaker_Query(t *testing.T) {
 	resp, err := c.Query(providers[0][0])
 	require.NoError(t, err)
 	require.GreaterOrEqual(t, len(resp.Offers), 1)
-	require.Equal(t, xmrmakerProvideAmount, resp.Offers[0].MinimumAmount)
-	require.Equal(t, xmrmakerProvideAmount, resp.Offers[0].MaximumAmount)
-	require.Equal(t, exchangeRate, float64(resp.Offers[0].ExchangeRate))
+	var respOffer *types.Offer
+	for _, offer := range resp.Offers {
+		if offerID == offer.ID.String() {
+			respOffer = offer
+		}
+	}
+	require.NotNil(t, respOffer)
+	require.Equal(t, xmrmakerProvideAmount, respOffer.MinimumAmount)
+	require.Equal(t, xmrmakerProvideAmount, respOffer.MaximumAmount)
+	require.Equal(t, exchangeRate, float64(respOffer.ExchangeRate))
```

Comment:
If you run this test multiple times, the new offer that we just added is in the last index, not the first index.  I wasn't sure if I should hard-code that though, so I search the the offer ID.  I was running these tests by hand out-of-order and the values `resp.Offers[0]` where not matching up, because the offer in index zero wasn't added by this test.